### ISSUE TITLE
Styles cannot be set for yaga-geojson (GeoJSONDirective)

### ISSUE THREAD

**Issue by AnatolyStrakhov** [yagajs/leaflet-ng2#316]
Hi,
I am using yaga-leaflet with packages:
```json
  "dependencies": {
    "angular/animations": "4.4.6",
    "angular/cdk": "2.0.0-beta.12",
    "angular/common": "4.4.6",
    "angular/compiler": "4.4.6",
    "angular/core": "4.4.6",
    "angular/forms": "4.4.6",
    "angular/http": "4.4.6",
    "angular/material": "2.0.0-beta.12",
    "angular/platform-browser": "4.4.6",
    "angular/platform-browser-dynamic": "4.4.6",
    "angular/router": "4.4.6",
    "yaga/leaflet-ng2": "1.0.0-rc9",
    "bootstrap": "3.3.7",
    "core-js": "2.4.1",
    "jquery": "3.1.1",
    "leaflet": "1.3.1",
    "ngx-bootstrap": "1.6.3",
    "ng2-cookies": "1.0.4",
    "openlayers": "4.0.1",
    "rxjs": "5.5.2",
    "ts-helpers": "1.1.1",
    "zone.js": "0.8.4"
  },
  "devDependencies": {
    "angular/cli": "1.6.2",
    "angular/compiler-cli": "4.4.6",
    "types/node": "6.0.42",
    "ts-node": "1.2.1",
    "tslint": "4.3.0",
    "typescript": "2.4"
  }
```

I cannot set neither [styler] nor [defaultStyle] for yaga-geojson. That's because of middleware is used in [https://github.com/yagajs/leaflet-ng2/blob/develop/ts/geojson.directive.ts](url)
Lines 174-197 are initialized before lines 300-345. As the result inputs are lost.
Could you fix it please?

**Comment by atd-schubert** [comment]
I don't think that there is a problem with the initialization, but there seems to be another problem with infinit loops, like described in #317.

**Comment by atd-schubert** [comment]
@AnatolyStrakhov can you take a look if the above mentioned commit fixes your bug and close this issue or give some feedback?

**Comment by AnatolyStrakhov** [comment]
@atd-schubert It seems the master branch has your commit in the develop branch. Or am I wrong?
`    public setData(val: GeoJSONFeatureCollection<GeometryObject, T>): this {
        super.clearLayers();
        super.addData(val);
        this.dataChange.emit((this.toGeoJSON() as GeoJSONFeatureCollection<GeometryObject, T>));
        return this;
    } `

I investigated the issue in depth and found that angular2 has an array of bindings of view's properties, and it goes through the array during initialization. 'data' has the first position, that's why values of other properties are lost.
I have yaga-leaflet installed with `npm install`. In `node_modules\@yaga\leaflet-ng2\lib\geojson.directive.js` I changed the order of GeoJSONDirective.propDecorators. I moved `'data': [{ type: core_1.Input },],` to the end of the list. It solves my problem, now I can set styles. Please find original and new files in zip.

[geojson.directive.zip](https://github.com/yagajs/leaflet-ng2/files/1677412/geojson.directive.zip)

The file is generated, not a source. But I am a beginner in Angular, so I cannot point exactly what must be fixed in sources to have this order of items. If you need any information from my side please let me know.

**Comment by atd-schubert** [comment]
Can you try to install the latest development version of leaflet-ng2 and check it again?

```bash
npm install --save github:yagajs/leaflet-ng2#develop
```

You can undo this change after the next release of leaflet-ng2

**Comment by AnatolyStrakhov** [comment]
I installed it successfully but it fails to run

`ERROR in ./node_modules/@yaga/leaflet-ng2/index.js
Module not found: Error: Can't resolve './lib/index' in '<my folder>\node_modules\@yaga\leaflet-ng2'
resolve './lib/index' in '<my folder>\node_modul
s\@yaga\leaflet-ng2'
  using description file: <my folder>\node_modul
s\@yaga\leaflet-ng2\package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: <my folder>\node
modules\@yaga\leaflet-ng2\package.json (relative path: .)
    using description file: <my folder>\node_mod
les\@yaga\leaflet-ng2\package.json (relative path: ./lib/index)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        <my folder>\node_modules\@yaga\leaflet-n
2\lib\index doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        <my folder>\node_modules\@yaga\leaflet-n
2\lib\index.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        <my folder>\node_modules\@yaga\leaflet-n
2\lib\index.js doesn't exist
      as directory
        <my folder>\node_modules\@yaga\leaflet-n
2\lib\index doesn't exist
[<my folder>\node_modules\@yaga\leaflet-ng2\lib\
ndex]
[<my folder>\node_modules\@yaga\leaflet-ng2\lib\
ndex.ts]
[<my folder>\node_modules\@yaga\leaflet-ng2\lib\
ndex.js]
[<my folder>\node_modules\@yaga\leaflet-ng2\lib\
ndex]
 @ ./node_modules/@yaga/leaflet-ng2/index.js 1:17-39
 @ ./src/app/app.module.ts
 @ ./src/app/index.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
ERROR in Error: YagaModule is not an NgModule
    at _getNgModuleMetadata (<my folder>\node_mo
ules\@angular\compiler-cli\src\ngtools_impl.js:140:15)
    at _extractLazyRoutesFromStaticModule (<my folder>\node_modules\@angular\compiler-cli\src\ngtools_impl.js:109:26)
    at <my folder>\node_modules\@angular\compile
-cli\src\ngtools_impl.js:129:27
    at Array.reduce (native)
    at _extractLazyRoutesFromStaticModule (<my folder>\node_modules\@angular\compiler-cli\src\ngtools_impl.js:128:10)
    at Object.listLazyRoutesOfModule (<my folder>\node_modules\@angular\compiler-cli\src\ngtools_impl.js:53:22)
    at Function.NgTools_InternalApi_NG_2.listLazyRoutes (<my folder>\node_modules\@angular\compiler-cli\src\ngtools_api.js:91:39
    at AotPlugin._getLazyRoutesFromNgtools (<my folder>\node_modules\@ngtools\webpack\src\plugin.js:241:66)
    at _donePromise.Promise.resolve.then.then.then.then.then (<my folder>\node_modules\@ngtools\webpack\src\plugin.js:495:24)
    at process._tickCallback (internal/process/next_tick.js:109:7)`

I use angular-cli and run it by `ng serve --proxy=proxy.conf.json`

**Comment by atd-schubert** [comment]
Oh yes sorry, it would not work because it is not transpiled with `ngc`! It is the "raw" library from GitHub. I forget this. You can transpile it on your own, but maybe it is a little bit tricky, because you need some dependencies for the transpile process, that you are not allowed to serve with that library as module.

For me it works now with the latest fixes. I will keep this issue open until the next official release. Then I will ping you in this issue to make the test again.

**Comment by AnatolyStrakhov** [comment]
@atd-schubert  When are you planning to have next release?

**Comment by atd-schubert** [comment]
There are some other fixes we want to include into the next release. I can not really make a promise of this, but I think it would be in the middle or end of February...

**Comment by TomWeber124** [comment]
Hello there. 

I´ve stumbled across the same issue as well.

Although you didn´t want to promise anything, middle/end of february has passed for a month by now and you´ve got me wondering about when you schedule the mentioned release? 

I don´t want to rush you though.
It´s been a pleasure using leaflet-ng2 so far. 
Thank you for your work!

**Comment by atd-schubert** [comment]
This issue is already merged in develop.

**Comment by AnatolyStrakhov** [comment]
Fixed in 1.0.0-rc10. Thank you.
